client/driver/rkt.go

```diff
@@ -288,23 +321,19 @@ func (h *rktHandle) Update(task *structs.Task) error {
 // Kill is used to terminate the task. We send an Interrupt
 // and then provide a 5 second grace period before doing a Kill.
 func (h *rktHandle) Kill() error {
-	h.proc.Signal(os.Interrupt)
+	h.executor.ShutDown()
 	select {
 	case <-h.doneCh:
 		return nil
 	case <-time.After(h.killTimeout):
-		return h.proc.Kill()
+		return h.executor.Exit()
 	}
 }
 
 func (h *rktHandle) run() {
-	ps, err := h.proc.Wait()
+	ps, err := h.executor.Wait()
 	close(h.doneCh)
-	code := 0
-	if !ps.Success() {
-		// TODO: Better exit code parsing.
-		code = 1
-	}
-	h.waitCh <- cstructs.NewWaitResult(code, 0, err)
+	h.waitCh <- &cstructs.WaitResult{ExitCode: ps.ExitCode, Signal: 0, Err: err}
```

Comment:
If the plugin errors out, we need to make sure the rkt container is killed, the allocdir cleaned up and cgroups cleaned up too.